### ISSUE TITLE
When you get civilian dependants as a reward through missions, it overflows the number of astechs and reduces astech hours to zero

### ISSUE THREAD

**Issue by Tamren** [MegaMek/mekhq#330]
MekHQ v42.0

Just like the title says. This happens when I take extraction raid missions, specifically the kind where you have to escort civilian vehicles to safety. One of the random rewards you can receive after the mission is civilian dependants who you can use as astechs. However if you already at maximum astech capacity this seems to confuse the game, when you advance a day the astech work pool drops to zero hours available. To fix the problem you have to release all temporary astechs and rehire them, then advance the day again. This returns your astech work hours to the normal amount.

This seems to be a problem specifically when the game adds the astechs you get from missions, because if you simply hire more astechs than you need nothing bad happens.

**Comment by pheonixstorm** [comment]
Do you have a campaign save from before and after this happens?

If not, before your next mission that gives bonuses make sure to save before and after. If you get more dependents then zip and attach the saves to this ticket.

**Comment by Tamren** [comment]
Yes I do have some saves. I looked back through my save archive and I managed to locate one before and after the problem. Again, this is using MekHQ 42.

[astec bug saves and custom units.zip](https://github.com/MegaMek/mekhq/files/818493/astec.bug.saves.and.custom.units.zip)

- I included my entire custom unit folder for this specific campaign, it probably won't load without it.
- C82 is the save made on June 3, there is an impending Extraction mission where I have to rescue 4 civilian vehicles.
- C83 is a save made on June 4 the day after the extraction was complete and I had received several dependants as my bonus reward. This one shows Astech work minutes as zero.

I did some basic testing on my own and here is what I found:

- I took save C82 and reran the extraction mission again, I didn't actually play out the mission in full, I simply deployed all units and then used /victory to end the game. This generated the "astech test campaign" MUL file. I then reloaded save C82 several times and used this MUL file to resolve the mission instantly until it rewarded me with dependants again, 6 to be specific. I then saved the game, this is save C82a.
- If I load up C82a and hit the button to advance the day, the Astech minutes pool instantly drops to zero, just like save C83. In order to fix the problem I have to fire all temporary Astechs and then rehire them.
- I thought it might have to do with work jobs that were in progress when the day ended, so I used the GM tools to finish all of the work in the repair bay and warehouse. This had no effect, astech minutes still became zero. I tried again and simply deleted all mechs and parts that were being worked on, same result.
- I tried using the GM tools to delete all of my dependants and then advanced the day. This had no effect, the astech minutes still became zero. However in this case all I had to do was bring all astech teams to full, I did not have to fire all of the temporary astechs first.
